src/Program.ts

```diff
@@ -326,15 +360,21 @@ export class Program {
                 let xmlFile = new XmlFile(pathAbsolute, pkgPath, this);
                 //add the file to the program
                 this.files[pathAbsolute] = xmlFile;
-                let fileContents = await getFileContents();
-                await xmlFile.parse(fileContents);
+                let fileContents: SourceObj = {
+                    pathAbsolute: pathAbsolute,
+                    source: await getFileContents()
+                };
+                this.plugins.emit('beforeFileParse', fileContents);
+                await xmlFile.parse(
+                    fileContents.source,
+                    () => this.plugins.emit('afterFileParse', xmlFile)
```

Comment:
`XmlFile` already has a reference to `Program`, so I don't think we need to pass in a callback function for parse, because the `XmlFile` can just call `this.program.plugins.emit('afterFileParse', this)`.